selenium/spec/asset_library_urls_spec.rb

```diff
@@ -0,0 +1,123 @@
+    @asset_library.click_add_url_button
+    @asset_library.wait_for_asset_in_gallery(@driver, 'URL Title 1')
+    expect(@asset_library.gallery_asset_owner_name_elements[0].text).to eql("by #{test_teacher['fullName']}")
+    @asset_library.click_asset_link 0
+    @asset_library.wait_for_asset_detail(@driver, 'URL Title 1')
+    # TODO: verify asset category, description, and embedded URL once visible in the viewport
+  end
+
+  it 'allows the user to add a URL, title, and category to the asset library' do
+    @asset_library.enter_url_metadata('en.wikipedia.org', 'URL Title 2', 'Category 1', nil)
+    @asset_library.click_add_url_button
+    @asset_library.wait_for_asset_in_gallery(@driver, 'URL Title 2')
+    expect(@asset_library.gallery_asset_owner_name_elements[0].text).to eql("by #{test_teacher['fullName']}")
+    @asset_library.click_asset_link 0
+    @asset_library.wait_for_asset_detail(@driver, 'URL Title 2')
+    # TODO: verify asset category and embedded URL once visible in the viewport
+  end
+
+  it 'allows the user to add a URL and title to the asset library' do
+    @asset_library.enter_url_metadata('en.wikipedia.org', 'URL Title 3', nil, 'URL description')
+    @asset_library.click_add_url_button
+    @asset_library.wait_for_asset_in_gallery(@driver, 'URL Title 3')
+    expect(@asset_library.gallery_asset_owner_name_elements[0].text).to eql("by #{test_teacher['fullName']}")
+    @asset_library.click_asset_link 0
+    @asset_library.wait_for_asset_detail(@driver, 'URL Title 3')
+    # TODO: verify asset embedded URL once visible in the viewport
+  end
+
+  it 'allows the user to add a URL, title, and description to the asset library' do
+    @asset_library.enter_url_metadata('en.wikipedia.org', 'URL Title 4', nil, 'URL description')
+    @asset_library.click_add_url_button
+    @asset_library.wait_for_asset_in_gallery(@driver, 'URL Title 4')
+    expect(@asset_library.gallery_asset_owner_name_elements[0].text).to eql("by #{test_teacher['fullName']}")
+    @asset_library.click_asset_link 0
+    @asset_library.wait_for_asset_detail(@driver, 'URL Title 4')
+    # TODO: verify asset description and embedded URL once visible in the viewport
+  end
+
+  it 'requires the user to enter a URL' do
+    @asset_library.enter_url_metadata(nil, 'URL Title 5', nil, nil)
+    @asset_library.click_add_url_button
+    @asset_library.missing_url_error_element.when_visible timeout=WebDriverUtils.page_update_wait
+    expect(@asset_library.url_title_input).to eql('URL Title 5')
+  end
+
+  it 'requires the user to enter a valid URL' do
+    @asset_library.enter_url_metadata('foo bar', 'URL Title 6', nil, nil)
+    @asset_library.click_add_url_button
+    @asset_library.bad_url_error_element.when_visible timeout=WebDriverUtils.page_update_wait
+    expect(@asset_library.url_input).to eql('http://foo bar')
+    expect(@asset_library.url_title_input).to eql('URL Title 6')
```

Comment:
@nicolaasmatthijs it actually is verifying the message, since the message element is defined by its text.